obspy/clients/fdsn/client.py

```diff
@@ -45,6 +46,7 @@
 
 DEFAULT_SERVICE_VERSIONS = {'dataselect': 1, 'station': 1, 'event': 1}
 
+URL_REGEX = 'https?://(localhost|\w+(\.\w+){2,})(:\d+)?(/\w+)*/?$'
```

Comment:
Schemes are technically case-insensitive, `\w` is `[a-zA-Z0-9_]` (plus other Unicode "word characters") but paths can include more than that, I think. Maybe it would be simpler to pass through `urlparse`?